csrc/iter_visitor.cpp

```diff
@@ -938,6 +938,98 @@ std::vector<Statement*> StmtSort::getStmtsTo(
   return es.stmts;
 }
 
+std::vector<Statement*> StmtSort::getAllStmts(
+    Fusion* fusion,
+    bool traverse_members,
+    bool traverse_attributes,
+    bool traverse_siblings) {
+  return getAllStmtsTo(
+      fusion->getTerminatingOutputs(),
+      traverse_members,
+      traverse_attributes,
+      traverse_siblings);
+}
+
+std::vector<Statement*> StmtSort::getAllStmtsTo(
+    const std::vector<Val*>& to,
+    bool traverse_members,
+    bool traverse_attributes,
+    bool traverse_siblings) {
+  // If members are not traversed, this can just be handled by getStmts
+  if (!traverse_members) {
+    return getStmtsTo(
+        to, traverse_members, traverse_attributes, traverse_siblings);
+  }
+
+  // to is assumed to include only scalar or TensorView
+  NVF_ERROR(std::all_of(to.begin(), to.end(), [](Val* to_val) {
+    return to_val->vtype() == ValType::TensorView ||
+        to_val->vtype() == ValType::Others;
+  }));
+
+  // First, grab all statements without traversing tensor members
+  auto stmts = getStmtsTo(to, false, traverse_attributes, traverse_siblings);
+
+  VectorOfUniqueEntries<Statement*> all_stmts;
+
+  // For TensorView, further traversing into its members. Note that
+  // traverse_members is always true here
```

Comment:
Since `traverse_members` is always considered `true` with respect to `TensorView`/`TensorDomain`, what effect does that parameter have now? I think now this just means "visit IterDomain start, stop, extent scalars"

Reply:
If the flag is true, the behavior is the same as `getStmtsTo` as it falls back to the function.

This is because I was only thinking about a typical use case of `StmtSort::getAllStmtsTo(fusion->outputs())`. In that case, if the flag is true, it should just return tensors and scalars, but no TensorDomain.

The only case the difference of getStmtsTo and getAllStmtsTo is when the `to` target includes TensorDomain. `getStmtsTo` only traverses from the loop domain to the root domain, whereas `getAllStmtsTo` uses `TensorDomain::getAllIStatements`.

This interface is far from ideal but good enough for the immediate problem I'm facing (fixing the replace size pass). Ideally, there should be only one "getStmtsTo". I'd suggest we'd start from here and incrementally update as we wish.

Reply:
Sounds good.